File: xlog/src/main/java/com/salton123/log/XLogConfig.java
package com.salton123.log;

/**
 * User: newSalton@outlook.com
 * Date: 2019/4/26 10:37 PM
 * ModifyTime: 10:37 PM
 * Description:
 */
public class XLogConfig {
    private boolean isDebugable = true;  //日志开关
    private String  savePath = Utils.getDefaultPath() ;  //保存位置
    private boolean whetherToSaveLog = false; //是否保存日志
    private int  logDeleteDelayDay = 10 ;  //默认10天后删除
    private int logDefaultSplitSize = 5;  //日志拆分大小阈值，每1M拆分

    public boolean isDebugable() {
        return isDebugable;
    }

    public String getSavePath() {
        return savePath;
    }

    public boolean isWhetherToSaveLog() {
        return whetherToSaveLog;
    }

    public int getLogDeleteDelayDay() {
        return logDeleteDelayDay;
    }

    public int getLogDefaultSplitSize() {
        return logDefaultSplitSize;
    }

    public XLogConfig setDebugable(boolean debugable) {
        isDebugable = debugable;
        return this;
    }

    public XLogConfig setSavePath(String savePath) {
        this.savePath = savePath;
        return this;
    }

    public XLogConfig setWhetherToSaveLog(boolean whetherToSaveLog) {
        this.whetherToSaveLog = whetherToSaveLog;
        return this;
    }

    public XLogConfig setLogDeleteDelayDay(int logDeleteDelayDay) {
        this.logDeleteDelayDay = logDeleteDelayDay;
        return this;
    }

    public XLogConfig setLogDefaultSplitSize(int logDefaultSplitSize) {
        this.logDefaultSplitSize = logDefaultSplitSize;
        return this;
    }
}

File: xlog/src/main/java/com/salton123/log/XLogConfig.kt
package com.salton123.log

/**
 * Time:2022/1/11 15:54
 * Author:
 * Description:
 * @param isDebugable 日志开关
 * @param whetherToSaveLog 是否保存日志
 * @param savePath 保存位置
 * @param logDeleteDelayDay 默认10天后删除
 * @param logDefaultSplitSize 日志拆分大小阈值，每1M拆分
 */
class XLogConfig {

    var isDebugable: Boolean = true
        private set

    var whetherToSaveLog: Boolean = true
        private set
    var savePath: String = ""
        private set
    var logDeleteDelayDay: Int = 10
        private set
    var logDefaultSplitSize: Int = 5
        private set
    var callback: ((code: Int, message: String) -> Unit)? = null
        private set

    fun setDebugable(debugable: Boolean): XLogConfig {
        isDebugable = debugable
        return this
    }

    fun setSavePath(savePath: String?): XLogConfig {
        this.savePath = savePath!!
        return this
    }

    fun setWhetherToSaveLog(whetherToSaveLog: Boolean): XLogConfig {
        this.whetherToSaveLog = whetherToSaveLog
        return this
    }

    fun setLogDeleteDelayDay(logDeleteDelayDay: Int): XLogConfig {
        this.logDeleteDelayDay = logDeleteDelayDay
        return this
    }

    fun setLogDefaultSplitSize(logDefaultSplitSize: Int): XLogConfig {
        this.logDefaultSplitSize = logDefaultSplitSize
        return this
    }

    fun setCallback(callback: ((code: Int, message: String) -> Unit)?): XLogConfig {
        this.callback = callback
        return this
    }
}